src/drivers/intel/ssp/ssp.c

```diff
@@ -67,22 +67,23 @@ static void ssp_empty_tx_fifo(struct dai *dai)
 /* empty SSP receive FIFO */
 static void ssp_empty_rx_fifo(struct dai *dai)
 {
+	struct timer *timer = timer_get();
+	uint64_t deadline = platform_timer_get(timer) +
+		clock_ms_to_ticks(PLATFORM_DEFAULT_CLOCK, 1) *
+		SSP_RX_FLUSH_TIMEOUT / 1000;
 	uint32_t sssr;
-	uint32_t entries;
-	uint32_t i;
 
-	sssr = ssp_read(dai, SSSR);
+	do {
+		while (ssp_read(dai, SSSR) & SSSR_RNE)
+			/* read to empty the fifo */
+			ssp_read(dai, SSDR);
+	} while ((ssp_read(dai, SSSR) & SSSR_BSY) &&
+		 platform_timer_get(timer) < deadline);
 
+	sssr = ssp_read(dai, SSSR);
```

Comment:
I find this code absolutely horrible. The last ssp_read doesn't seem necessary and the inner loop with no limit looks like a bug.

Reply:
My point exactly:

````c
do {
		while (pxa2xx_spi_read(drv_data, SSSR) & SSSR_RNE)
			pxa2xx_spi_read(drv_data, SSDR);
	} while ((pxa2xx_spi_read(drv_data, SSSR) & SSSR_BSY) && --limit);
	write_SSSR_CS(drv_data, SSSR_ROR);
````

Why would you need *another read* after the 'while'?

Reply:
> My point exactly:
> 
> ```c
> do {
> 		while (pxa2xx_spi_read(drv_data, SSSR) & SSSR_RNE)
> 			pxa2xx_spi_read(drv_data, SSDR);
> 	} while ((pxa2xx_spi_read(drv_data, SSSR) & SSSR_BSY) && --limit);
> 	write_SSSR_CS(drv_data, SSSR_ROR);
> ```
> 
> Why would you need _another read_ after the 'while'?

Oh, makes sense, let me align it to kernel one and just clear the interrupt bit blindly.

done.